File: NorthwndBackend/NorthwndBackend/Dtos/OrderDetailDto.cs
namespace NorthwndBackend.Dtos;

public class OrderDetailDto
{
    public int OrderId { get; set; }
    public short Quantity { get; set; }
    public string Product { get; set; }
    public string Category { get; set; }
    public decimal UnitPrice { get; set; }
}

File: NorthwndFrontend/src/data/orderDetailDto.ts
export class OrderDetailDto {
    orderId: number;
    quantity: number;
    unitPrice: number;
    productId: number;
    product?: string;
    category?: string;

    constructor(orderId: number, quantity: number, unitPrice: number, productId: number, product: string, category: string) {
        this.orderId = orderId;
        this.quantity = quantity;
        this.unitPrice = unitPrice;
        this.productId = productId;
        this.product = product;
        this.category = category;
    }
}